engine/include/cubos/engine/input/bindings.hpp

```diff
@@ -0,0 +1,115 @@
+/// @file
+/// @brief
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <cubos/core/io/keyboard.hpp>
+
+namespace cubos::engine
+{
+    class KeyWithModifier final
+    {
+    public:
+        KeyWithModifier() = default;
+        KeyWithModifier(cubos::core::io::Key key, cubos::core::io::Modifiers modifiers)
+            : mKey(key)
+            , mModifiers(modifiers)
+        {
+        }
+
+        ~KeyWithModifier() = default;
+
+        const cubos::core::io::Key& getKey() const;
+        const cubos::core::io::Modifiers& getModifiers() const;
+
+        cubos::core::io::Key& getKey();
+        cubos::core::io::Modifiers& getModifiers();
+
+        std::string toString() const;
+
+    private:
+        cubos::core::io::Key mKey;
+        cubos::core::io::Modifiers mModifiers;
+    };
+
+    class Axis final
+    {
+    public:
+        Axis() = default;
+        Axis(std::vector<KeyWithModifier> positive, std::vector<KeyWithModifier> negative)
+            : mPositive(positive)
+            , mNegative(negative)
+        {
+        }
+
+        ~Axis() = default;
+
+        const std::vector<KeyWithModifier>& getPositive() const;
+        const std::vector<KeyWithModifier>& getNegative() const;
+
+        std::vector<KeyWithModifier>& getPositive();
+        std::vector<KeyWithModifier>& getNegative();
+
+        float getValue() const;
+        void setValue(float value);
```

Comment:
Use `value()` and `value(float value)` for consistency (the same goes for other methods).